src/main/scala/scorex/api/http/BlocksApiRoute.scala

```diff
@@ -79,8 +80,14 @@ case class BlocksApiRoute(settings: RestAPISettings, checkpointsSettings: Checkp
     new ApiImplicitParam(name = "signature", value = "Base58-encoded signature", required = true, dataType = "string", paramType = "path")
   ))
   def heightEncoded: Route = (path("height" / Segment) & get) { encodedSignature =>
-    withBlock(history, encodedSignature) { block =>
-      complete(Json.obj("height" -> history.heightOf(block)))
+    if (encodedSignature.length > TransactionParser.SignatureStringLength)
```

Comment:
We should probably come up with a better utility method for extracting signatures from request URIs.